internal/provider/framework/config.go

```diff
@@ -0,0 +1,642 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-sdk/sdk/auth"
+	"github.com/hashicorp/go-azure-sdk/sdk/environments"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	providerfeatures "github.com/hashicorp/terraform-provider-azurerm/internal/features"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/provider"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/resourceproviders"
+)
+
+type ProviderConfig struct {
+	clientBuilder clients.ClientBuilder
+	Client        *clients.Client
+}
+
+// Load handles the heavy lifting of configuring the provider and handling defaults
+func (p *ProviderConfig) Load(ctx context.Context, data *ProviderModel, tfVersion string, diags *diag.Diagnostics) {
+	var (
+		env *environments.Environment
+		err error
+	)
+
+	if metadataHost := data.MetaDataHost.ValueString(); metadataHost != "" {
```

Comment:
we should add a comment around how we handle the metadatahost and env selection process here?